src/main/java/net/dv8tion/jda/internal/managers/AudioManagerImpl.java

```diff
@@ -328,30 +317,23 @@ public ConnectionListener getListenerProxy()
 
     public void setAudioConnection(AudioConnection audioConnection)
     {
-        this.audioConnection = audioConnection;
         if (audioConnection == null)
+        {
+            this.audioConnection = null;
             return;
+        }
 
-        this.queuedAudioConnection = null;
+        // This will set the audioConnection to null, which we then immediately override with the new connection
+        if (this.audioConnection != null)
+            closeAudioConnection(ConnectionStatus.AUDIO_REGION_CHANGE);
```

Comment:
So we are just assuming that if we had an audio connection already and we're changing to a new one that it is a region change?

So long as `setAudioConnection` is called with a non-null value _only_ from `VOICE_SERVER_UPDATE`, I suppose that is fine.

Reply:
Yeah, every time we call `setAudioConnection` it's because of a region/server change. "Region" in this case basically just means we switched voice server. One guild can now have multiple voice servers for individual channels, moving channel would then "change region". It is an unfortunate naming of this status but we'll keep using it.

Reply:
> One guild can now have multiple voice servers for individual channels, moving channel would then "change region". It is an unfortunate naming of this status but we'll keep using it.

Wow really? Interesting.